django/db/backends/mysql/base.py

```diff
@@ -360,6 +360,8 @@ def check_constraints(self, table_names=None):
                     )
 
     def is_usable(self):
+        if self.connection is None:
+            return False
```

Comment:
See my comment from the ticket:

> Also, rather than checking for `connection is None` in advance, using [https://docs.python.org/3/glossary.html#term-eafp EAFP] and checking for `None` in an `except` block (e.g. handling AttributeError) is probably more Pythonic.